modules/site-health/audit-database/class-perflabdbtests.php

```diff
@@ -0,0 +1,435 @@
+<?php
+/**
+ * Performance Lab database test module
+ *
+ * @package performance-lab
+ * @group audit-database
+ *
+ * @since 1.4.0
+ */
+
+/**
+ * Performance Lab database test class.
+ *
+ * This class declares and runs the database performance auditing tests.
+ * It uses a passed-in PerflabDbMetrics instance to access the database.
+ *
+ * @package performance-lab
+ * @group audit-database
+ *
+ * @since 1.4.0
+ */
+class PerflabDbTests {
+	/** Metrics-retrieving instance.
+	 *
+	 * @var object Collection of functions.
+	 */
+	private $metrics;
+	/** Database server semantic version and capability data.
+	 *
+	 * @var object Version descriptor.
+	 */
+	private $version;
+	/** Database server name.
+	 *
+	 * @var string MySQL or MariaDB.
+	 */
+	private $name;
+	/** Skip all tests.
+	 *
+	 * @var bool true if we should skip all tests
+	 */
+	private $skip_all_tests = false;
+	/** Thresholds instance.
+	 *
+	 * @var PerflabDbUtilities instance.
+	 */
+	private $utilities;
+	/** Table Information.
+	 *
+	 * @var array associative array of table information.
+	 */
+	private $table_stats;
+
+	/** Results accumulator.
+	 *
+	 * @var [string]
+	 */
+	private $results = array();
+
+	/** Constructor for tests.
+	 *
+	 * @param PerflabDbMetrics $metrics Metrics-retrieving instance.
+	 */
+	public function __construct( $metrics ) {
+
+		$this->metrics     = $metrics;
+		$this->version     = $this->metrics->get_db_version();
+		$this->name        = $this->version->server_name;
+		$this->utilities   = PerflabDbUtilities::get_instance();
+		$this->table_stats = $this->metrics->get_table_info();
+	}
+
+	/** Add all Site Health checks for database performance.
+	 *
+	 * @param array $tests Pre-existing tests.
+	 *
+	 * @return array Augmented tests.
+	 */
+	public function add_all_database_performance_checks( $tests ) {
+		$tests['direct']['database_performance'] = array(
+			'label' => __( 'Database', 'performance-lab' ),
+			'test'  => array( $this, 'alltests' ),
+		);
+		return $tests;
+	}
+
+	/** Accumulate results, tracking status.
+	 *
+	 * @param array $result Test result from individual test.
+	 *
+	 * @return void
```

Comment:
Please omit any `@return void`.
```suggestion
	 * @param array $result Test result from individual test.
```